### ISSUE TITLE
Add support for icon font

### ISSUE THREAD

**Issue by ilPas** [gsoft-inc/storybook-icon-gallery#9]
### Is your feature request related to a problem?

I have in my Storybook an icon font for my personal font and the variants doesn't work
It's possibile add the support?

### Describe the solution you'd like

Add support for variants

**Comment by patricklafrance** [comment]
Hi @ilPas 

Since an `<IconGallery.Variant />` accept a React component you can provide a React component that use your ico font.

It should work out of the box.

**Comment by ilPas** [comment]
Yes but if i want set a variant it doesn't work with icon font because change the width of icon container and not the font size

**Comment by patricklafrance** [comment]
Your custom icon component will receive a width and height matching the variant size, from there, can't you make sure you render the right classes to match the variant size?

**Comment by ilPas** [comment]
I use the variant, that change width and height but not the font size... i must create a props for my custom icon component?!

**Comment by patricklafrance** [comment]
Maybe I don't understand the issue correctly, I haven't use icofont for a while.

The way I see it, you could have the following code:

```
<IconGallery.Variants name="add">
    <IconGallery.Variant size={24}>
        <AddIcon24 />
    </IconGallery.Variant>
</IconGallery.Variants>
```

`<AddIcon24 />` is your custom component, which mean that it can render any icofont that fit a 24x24 variant.

**Comment by ilPas** [comment]
Sorry for my bad english.

` <IconGallery.Item name="lavoro">
          <IconGallery.Variant size={16}>
            <Icona>lavoro</Icona>
          </IconGallery.Variant>
        </IconGallery.Item>`

This is my code and the componente Icona is:

`const Icona = (props) => {
  return <Icon className="Hype-icon" {...props} />;
};`

If i change the Variant the font size doesn't change automatically.
The only way is create  props in my custom component and not use the Variant?

**Comment by patricklafrance** [comment]
You're right, the font-size will not change automatically, this is not how this component work and I am not sure I am willing to change it to support this.

You could do something like:

```
const Icona16 = ({ style, ...props }) => { return <Icon className="Hype-icon font-size-16" {...props} />; };
```
You might not need to exclude to `style` prop.. A variant component will provide a `width` and `height` matching the variant size.

**Comment by ilPas** [comment]
Yes, thanks.
Just in case...if i create a icon font support in your component can i make a pull request?

Thanks a lot

**Comment by patricklafrance** [comment]
It depends, what do you have in mind?

**Comment by ilPas** [comment]
I don't know but if i work on it next weeks i can think someting and propose you

**Comment by patricklafrance** [comment]
Sure, I can't guarantee it will be merge though, I want to keep this component simple.

So far, the way I see it is that using an SVG, IcoFont, or any other icon format is in user land.

I might be wrong.
